src/cpp/src/visual_language/clip.cpp

```diff
@@ -76,7 +76,12 @@ void bicubic_resize(const clip_image_u8 &img, clip_image_u8 &dst, int target_wid
 
     dst.nx = target_width;
     dst.ny = target_height;
-    dst.buf.resize(3 * target_width * target_height);
+    const size_t target_size = static_cast<size_t>(3 * target_width * target_height);
+    dst.buf.resize(target_size);
+    if (img.nx == target_width && img.ny == target_height) {
+        std::memcpy(dst.buf.data(), img.buf.data(), target_size*sizeof(uint8_t));
+        return;
```

Comment:
why can't we just return or use `img.buf.data()` without this extra copy? For this case, it is quite strange to do such full copy. Let use img as is

Reply:
I just want to avoid below complex calculation if target and source image have same height and width.

Actually, because `bicubic_resize `is called in many places,  in order to avoid to introduce new potential issue in the future, so I use this deep copy.

Reply:
Yes. removed.